specification/ai/Azure.AI.Projects/datasets/routes.tsp

```diff
@@ -51,11 +52,23 @@ interface Datasets
       ListDatasetVersionsParameters
     > {
   /**
-   * Create or start a pending upload of a dataset
+   * Create or start a pending upload of a dataset for a specific version
    */
   startPendingUpload is ServicePatterns.VersionedResourceAction<
```

Comment:
Is it intentional that there is a difference between the two ServicePatterns used for these two operations?
  startPendingUpload                         is ServicePatterns.VersionedResourceAction<>
  startPendingUploadAutoIncrement is ServicePatterns.BuildingBlocks.RepeatableCoreOps.ResourceAction<>

Reply:
Yes, the versioned service pattern is used for any resource actions under `/versions`

The one from `BuildingBlocks` operates at the top-level resource (i.e `/datasets`)